src/main/java/io/nash/openlimits/NashClient.java

```diff
@@ -0,0 +1,94 @@
+ * This is neccessary because the library is actually using websockets to make the requests and requires recreating the client if connection is broken
+ * When Nash adds an auto-reconnect on the Rust web sockets we should be able to remove this wrapper
+ */
+public class NashClient {
+    NashCredentials credentials;
+    ExchangeClient client;
+
+    private void buildClient() {
+        client = new ExchangeClient(
+                    new ExchangeClientConfig(
+                            new NashConfig(
+                                    credentials,
+                                    0,
+                                    "production",
+                                    10000,
+                                    "2PTzyS"
+                            )
+                    )
+        );
+    }
+
+    public NashClient() {
+        buildClient();
+    }
+
+    public NashClient(String apiKey, String secretKey) {
+        credentials = new NashCredentials(secretKey, apiKey);
+        buildClient();
+    }
+
+    public Order limitBuy(LimitRequest request) {
+        return wrapCall((Callable<Order>) () -> client.limitBuy(request));
+    }
+
+    public Order limitSell(LimitRequest request) {
+        return wrapCall((Callable<Order>) () -> client.limitSell(request));
+    }
+
+    public Order marketBuy(MarketRequest request) {
+        return wrapCall((Callable<Order>) () -> client.marketBuy(request));
+    }
+
+    public Order marketSell(MarketRequest request) {
+        return wrapCall((Callable<Order>) () -> client.marketSell(request));
+    }
+
+    public Balance[] getAccountBalances(Paginator paginator) {
+        return wrapCall((Callable<Balance[]>) () -> client.getAccountBalances(paginator));
+    }
+    public OrderCanceled cancelOrder(CancelOrderRequest req) {
+        return wrapCall((Callable<OrderCanceled>) () -> client.cancelOrder(req));
+    }
+
+    public Order getOrder(GetOrderRequest req) {
+        return wrapCall((Callable<Order>) () -> client.getOrder(req));
+    }
+
+    public MarketPair[] receivePairs() {
+        return wrapCall((Callable<MarketPair[]>) () -> client.receivePairs());
+    }
+
+    public  Candle[] getHistoricRates(GetHistoryRatesRequest req) {
+        return wrapCall((Callable<Candle[]>) () -> client.getHistoricRates(req));
+    }
+
+    private <T> T wrapCall(Callable<T> callable) {
+        while(true) {
+            try {
+                return callable.call();
+            }
+            catch(NashProtocolError error) {
+                if (error.getMessage().contains("Could not register request with broker")) {
+                    buildClient();
+                    try {
+                        Thread.sleep(1000);
+                    } catch (InterruptedException e) {
+                    }
+                }
+                else {
+                    throw error;
+                }
+            }
+            catch(Exception error) {
+
```

Comment:
Can we avoid empty catch? Perhaps log or rethrow?

Reply:
Is it alright to just log to not have to throw Exception in all these methods?

Reply:
If the caller is expecting a return, how does silently ignoring the exception work? Can we just return or throw?

Reply:
The Exception will be the one from Callable and it will keep calling in an infinite loop, but the exceptions coming from the client calls are already handled

Reply:
Ah, I missed the `while(true)` initially. Ideally, we want to always have an exit condition, either number of retries or time limit, then if we still get exception after that limit, we can throw the last exception received. 
For consistency, the retry logic should also live outside the library, ideally we just throw specific exceptions that we understand here, then the caller can decide on retries, so that we have measurement and monitoring when this happens for all exchanges.